docs/architecture/devops-for-aspnet-developers/actions-deploy.md

```diff
@@ -0,0 +1,491 @@
+          uses: actions/download-artifact@v2.0.8
+          with:
+            name: website
+            path: website
+        
+        - name: Login via Azure CLI
+          uses: azure/login@v1
+          with:
+            creds: ${{ secrets.AZURE_CREDENTIALS }}
+        
+        - name: Deploy web app
+          id: deploywebapp
+          uses: azure/webapps-deploy@v2
+          with: 
+            app-name: ${{ env.app-name }}
+            slot-name: staging
+            package: website
+        
+        - name: az cli logout
+          run: az logout
+    ```
+
+    Notice the following things:
+
+    1. You're creating a new job called `deploy_staging`.
+    1. You specify a dependency using `needs`. This job needs the `build` job to complete successfully before it starts.
+    1. This job also runs on the latest Ubuntu hosted agent, as specified with the `runs-on` attribute.
+    1. You specify that this job is targeting the `PRE-PROD` environment using the `environment` object. You also specify the `url` property. This URL will be displayed in the workflow diagram, giving users an easy way to navigate to the environment. The value of this property is set as the `output` of the `step` with `id` `deploywebapp`, which is defined below.
+    1. You're executing a `download-artifact` step to download the artifact (compiled web app) from the `build` job.
+    1. You then `login` to Azure using the `AZURE_CREDENTIALS` secret you saved earlier. Note the `${{ }}` notation for dereferencing variables.
+    1. You then perform a `webapp-deploy`, specifying the `app-name`, `slot-name`, and path to the downloaded artifact (`package`). This action also defines an `output` parameter that you use to set the `url` of the environment above.
+    1. Finally, you execute a `logout` to logout of the Azure context
+
+1. Commit the file.
+1. When the run completes, you should see two successful jobs. The URL for the `PRE-PROD` stage has been set and selecting it will navigate you to your web app staging slot:
+
+    - ![Deployment to PRE-PROD is successful](./media/actions/deploy/deploy-to-staging-completed.jpg)
+    **Figure 6**: Deployment to PRE-PROD is successful.
+
+1. Notice how the URL contains `-staging` in the name since this is the direct URL to the staging slot:
+
+    - ![The staging slot running](./media/actions/deploy/deployed-to-staging.jpg)
+    **Figure 7**: The staging slot running.
+
+1. You can also now see deployments. Navigate to `https://{your repository url}/deployments` to view your deployments:
+
+    - ![View deployments](./media/actions/deploy/deployments.jpg)
+    **Figure 8**: View deployments.
+
+## Deploy to production
+
+Now that you've deployed successfully to `PRE-PROD`, you'll want to deploy to `PROD`. Deployment to `PROD` will be slightly different since you don't need to copy the website again - you just need to swap the `staging` slot with the production slot. you'll do this using an Azure CLI (`az`) command.
```

Comment:
```suggestion
Now that you've deployed successfully to `PRE-PROD`, you'll want to deploy to `PROD`. Deployment to `PROD` will be slightly different since you don't need to copy the website again. You just need to swap the `staging` slot with the production slot. You'll do this using an Azure CLI (`az`) command.
```